valet-key/bicep/main.bicep

```diff
@@ -60,7 +60,7 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2023-01-01' = {
     accessTier: 'Hot'
     allowBlobPublicAccess: false
     allowCrossTenantReplication: false
-    allowSharedKeyAccess: true // This storage account is configured to allow SaS tokens, which will be generated by the Azure Function and provided to clients as valet keys. 
+    allowSharedKeyAccess: false // Only managed identity allowed, we needed to change the way to generate SAS token using UserDelegationKey
```

Comment:
Um, but the shared access key is used to access the storage account as part of this.  How does disabling this work?  We need shared access keys for this scenario... this scenario is all about shared access keys -- it's a specifically for scenario where client managed identities do not work for the situation.

Reply:
I tested after disabled and it is working.  The implemetation use manage identity and the SAS token is generated by UserDelegationKey (so it is not needed for that). The Azure funtion don't need the access key to work (it used URL https://STORAGE_ACCOUNT_NAME.blob.core.windows.net and Role assigments), and the client work only with the SAS token, it doens't matter how it was created. 
I don't see the point why it should be needed.

Reply:
The client should NOT be using managed identity.  The whole scenario is specifically for a situation where client cann't use an identity from the IdP.  Your finding is very concerning.

Reply:
the client use the SAS token, it doesn't use managed Identity. It use the SAS Token generated by the Azure Funtion (The web part).

Reply:
Ahhhh, TIIL that SaS tokens still work with this disabled for Entra-signed tokens. Awesome.  Thanks!!

Reply:
https://learn.microsoft.com/en-us/azure/storage/common/shared-key-authorization-prevent?tabs=portal#understand-how-disallowing-shared-key-affects-sas-tokens explained this.